src/SIL.XForge.Scripture/Services/ParatextSyncRunner.cs

```diff
@@ -1668,6 +1668,16 @@ private void Log(string message, string? projectSFId = null, string? userId = nu
             LogMetric(message);
         }
 
-        private void LogMetric(string message) => _syncMetrics.Log.Add($"{DateTime.UtcNow} {message}");
+        private void LogMetric(string message)
+        {
+            _syncMetrics.Log.Add($"{DateTime.UtcNow} {message}");
+            _syncMetricsRepository
+                .ReplaceAsync(_syncMetrics, true)
```

Comment:
As this is not called with an `await`, there is a risk that this may be executed out of order, or that the async state might get messed up. Depending on how the MongoDB library uses the async state, it may cause bigger issues than just out of order executions. It probably won't...but I can't tell without heavy testing.

Is there a reason you didn't make this method an `async Task`?